src/Aspire.Hosting/DistributedApplicationBuilder.cs

```diff
@@ -102,6 +102,41 @@ public DistributedApplicationBuilder(string[] args) : this(new DistributedApplic
     // values in various callbacks and is a central location to access useful services like IServiceProvider.
     private readonly DistributedApplicationExecutionContextOptions _executionContextOptions;
 
+    private DistributedApplicationExecutionContextOptions BuildExecutionContextOptions()
+    {
+        var operationConfiguration = _innerBuilder.Configuration["AppHost:Operation"];
+        if (operationConfiguration is null)
+        {
+            return _innerBuilder.Configuration["Publishing:Publisher"] switch
+            {
+                { } publisher => new DistributedApplicationExecutionContextOptions(DistributedApplicationOperation.Publish, publisher),
+                _ => new DistributedApplicationExecutionContextOptions(DistributedApplicationOperation.Run)
+            };
+        }
+
+        var operation = _innerBuilder.Configuration["AppHost:Operation"]?.ToLowerInvariant() switch
+        {
+            "publish" => DistributedApplicationOperation.Publish,
+            "inspect" => DistributedApplicationOperation.Inspect,
+            "run" => DistributedApplicationOperation.Run,
+            _ => throw new DistributedApplicationException("Invalid operation specified. Valid operations are 'publish', 'inspect', or 'run'.")
+        };
+
+        if (operation == DistributedApplicationOperation.Run)
```

Comment:
Missed opportunity to employ pattern matching again (for computing return value)? 😆

Reply:
Updated to a switch statement just for you :)